code/mobs/mob.dm

```diff
@@ -112,6 +113,12 @@
 	else
 		return FALSE
 
+/mob/proc/equip_new(path)
+	var/I = new path
+	pick_up(I)
+	equip(I)
+	update_stats()
```

Comment:
`equip` calls `update_stats`, this shouldn't be necessary.

Reply:
that means there is a bug elsewhere and that has to be fixed. You shouldn't add hacks to hide bugs like this.